SimPEG/EM/Static/Utils/SurveyDesign.py

```diff
@@ -0,0 +1,439 @@
+            RxLoc = np.c_[xElec[:-1], xElec[1:]]
+            nSrc = SrcLoc.shape[0]
+            nRx = RxLoc.shape[0]
+            SrcID = []
+            RxID = []
+            nLeg = []
+            for iSrc in range(nSrc-2):
+                if nSrc - iSrc - 2 > nSpacing:
+                    nSounding = nSpacing
+                else:
+                    nSounding = nSrc - iSrc - 2
+                SrcID.append(np.ones(nSounding, dtype=int) * iSrc)
+                RxID.append(np.arange(nSounding) + iSrc + 1)
+                nLeg.append(np.arange(nSounding)+1)
+            # Set actual number of source pole
+            self.nSrc = nSrc-2
+
+        if surveyType == "dipole-pole":
+            SrcLoc = np.c_[xElec, xElec]
+            RxLoc = np.c_[xElec[:-1], xElec[1:]]
+            nSrc = SrcLoc.shape[0]
+            nRx = RxLoc.shape[0]
+            SrcID = []
+            RxID = []
+            nLeg = []
+            for iSrc in range(nSrc-2):
+                if nSrc - iSrc - 2 > nSpacing:
+                    nSounding = nSpacing
+                else:
+                    nSounding = nSrc - iSrc - 2
+                SrcID.append(np.ones(nSounding, dtype=int) * abs(iSrc-nSrc+1))
+                RxID.append(abs(np.arange(nSounding) + iSrc + 1-nRx+1))
+                nLeg.append(np.arange(nSounding)+1)
+            self.nSrc = nSrc-2
+
+        if surveyType == "pole-pole":
+            SrcLoc = np.c_[xElec, xElec]
+            RxLoc = np.c_[xElec, xElec]
+            nSrc = SrcLoc.shape[0]
+            nRx = RxLoc.shape[0]
+            SrcID = []
+            RxID = []
+            nLeg = []
+            for iSrc in range(nSrc-1):
+                if nSrc - iSrc - 1 > nSpacing:
+                    nSounding = nSpacing
+                else:
+                    nSounding = nSrc - iSrc - 1
+                SrcID.append(np.ones(nSounding, dtype=int) * iSrc)
+                RxID.append(np.arange(nSounding) + iSrc +1)
+                nLeg.append(np.arange(nSounding)+1)
+            self.nSrc = nSrc-1
+
+        self.SrcID = np.hstack(SrcID)
+        self.RxID = np.hstack(RxID)
+        self.nLeg = np.hstack(nLeg)
+        self.SrcLoc = SrcLoc
+        self.RxLoc = RxLoc
+        self.xElec = xElec
+
+    def plot2Dgeometry(self, iSrc=0, showIt=True):
+        if showIt:
+            fig = plt.figure(figsize=(8, 3))
+            xloc = (
+                self.SrcLoc[self.SrcID, 0] + self.SrcLoc[self.SrcID, 1] +
+                self.RxLoc[self.RxID, 0] + self.RxLoc[self.RxID, 1]
+                ) * 0.25
+            zloc = self.nLeg
+            x_temp = (
+                self.SrcLoc[self.SrcID[self.SrcID == iSrc], 0] +
+                self.SrcLoc[self.SrcID[self.SrcID == iSrc], 1] +
+                self.RxLoc[self.RxID[self.SrcID == iSrc], 0] +
+                self.RxLoc[self.RxID[self.SrcID == iSrc], 1]
+                ) * 0.25
+            z_temp = self.nLeg[self.SrcID == iSrc]
+            plt.plot(
```

Comment:
looks like you are repeating this quite a bit. could you turn it into a function?